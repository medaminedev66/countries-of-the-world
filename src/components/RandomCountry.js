import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router';
import { PropTypes } from 'prop-types';
import loading from '../loading.png';
import { selectCountry } from '../redux/countries/countries';

const RandomCountry = (props) => {
  const dispatch = useDispatch();
  const history = useHistory();
  const { countries } = props;
  const randomCountry = countries[Math.floor(Math.random() * countries.length)];
  const renderRedirect = (name) => {
    dispatch(selectCountry(name));
    history.push('/react-capstone/details');
  };

  return (
    <div
      className="Random-country"
      onClick={() => renderRedirect(randomCountry.name)}
      aria-hidden="true"
      data-testid="Random-country"
    >
      <div className="country-select">
        <img
          className="flag"
          src={randomCountry ? randomCountry.flags.png : loading}
          alt="flag"
        />
        <h2>{randomCountry ? randomCountry.name : 'Picking...'}</h2>
      </div>
      <p>
        Let&apos;s discover&nbsp;
        <span>{randomCountry ? randomCountry.name : '?'}</span>
      </p>
    </div>
  );
};

RandomCountry.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  countries: PropTypes.array.isRequired,
};
export default RandomCountry;
