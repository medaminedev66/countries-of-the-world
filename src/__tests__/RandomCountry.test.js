import React from 'react';
import RandomCountry from '../components/RandomCountry';
import { ReduxRenderer, renderWithRedux } from './Mocks/reduxStore';

describe('test the RandomCountry component', () => {
  const countries = [{ name: 'Morocco', flags: 'mr' }];
  it('matches the snapshot', () => {
    const tree = ReduxRenderer(
      <RandomCountry countries={countries} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { queryByTestId } = renderWithRedux(
      <RandomCountry countries={countries} />,
    );
    expect(queryByTestId('Random-country')).toBeTruthy();
  });
});
