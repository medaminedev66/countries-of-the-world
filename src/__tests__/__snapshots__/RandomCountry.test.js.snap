// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the RandomCountry component matches the snapshot 1`] = `
<div
  aria-hidden="true"
  className="Random-country"
  data-testid="Random-country"
  onClick={[Function]}
>
  <div
    className="country-select"
  >
    <img
      alt="flag"
      className="flag"
    />
    <h2>
      Morocco
    </h2>
  </div>
  <p>
    Let's discoverÂ 
    <span>
      Morocco
    </span>
  </p>
</div>
`;
